packages/router/router.ts

```diff
@@ -1981,33 +1981,39 @@ export function createRouter(init: RouterInit): Router {
       return;
     }
 
-    if (deletedFetchers.has(key)) {
-      updateFetcherState(key, getDoneFetcher(undefined));
-      return;
-    }
```

Comment:
This was too aggressive.  It works for `handleFetcherLoader` because there's nothing left to do for an unmounted fetcher.  For submissions we only want to short circuit for error/redirects - we need to let success results flow through so revalidation happens.

Reply:
Easiest to see the changes with whitespace ignored